### DISCUSSION TITLE
Yocto setup for carrier board (Jetson AGX Orin)

### DISCUSSION THREAD

**Discussion by spb06** [OE4T/meta-tegra#1646] [Q&A]
Hi there, 

We are looking to setup Yocto for a vendor carrier board that is running the Jetson AGX Orin SOM (same as the Jetson AGX Orin Devkit). We followed the instructions in the "[Create a Custom Machine](https://github.com/OE4T/meta-tegra/wiki/Creating-a-custom-MACHINE#custom-device-tree)". Specifically, we created a new machine config and flash config as advised in the wiki. 

For the machine conf file, we copied over the [jetson-agx-orin-devkit.conf](https://github.com/OE4T/meta-tegra/blob/master/conf/machine/jetson-agx-orin-devkit.conf) and renamed it to match our machine name. The carrier board vendor provided their own .DTB file that we believe should replace the .DTB file specified in [jetson-agx-orin-devkit.conf](https://github.com/OE4T/meta-tegra/blob/master/conf/machine/jetson-agx-orin-devkit.conf) but we are not sure the best way to go about doing that. 

Any guidance will be greatly appreciated!

**Comment by joekale-pp** [comment]
Hello,

You can add your devicetree files to the l4t kernel as a patch in the kernel recipe or to your own fork and as long as it compiles when the kernel for that machine is compiled you would only need to set the `KERNEL_DEVICETREE` variable in your machine conf to be that device tree name.

**Reply by spb06** [reply]
Hey there, 

Would you be able to point to an example where this is carried out? 

Is [this](https://www.digikey.com/en/maker/projects/intro-to-embedded-linux-part-5-how-to-enable-i2c-in-the-yocto-project/6843bbf9a83c4c96888fccada1e7aedf) relevant to what you described?
